### ISSUE TITLE
Transaction not commits when transfering from one account to another ManagerActiveRecord

### ISSUE THREAD

**Issue by uniqby** [yii2tech/balance#4]
When we call ManagerDbTransaction::transfer 

```
\yii2tech\balance\ManagerDbTransaction::beginDbTransaction
```

 called multiple times (in transfer, increase and decrease methods).

First time it's work normally and begins transaction, but in next times it's returns null, so we lost transaction object and cant' commit them

**Comment by klimov-paul** [comment]
This is expected: the transaction is commited only once at the top layer.

**Comment by uniqby** [comment]
Transaction not commited, so no changes presents in DB

**Comment by klimov-paul** [comment]
Unable to reproduce: have just tested it with MySQL InnoDB - `transfer()` works fine.

**Comment by uniqby** [comment]
https://github.com/yii2tech/balance/blob/master/ManagerActiveRecord.php#L154

probably should be `return $db->getTransaction(); ?`

**Comment by klimov-paul** [comment]
No

**Comment by AlfioSaitta** [comment]
Same problem for my project....

My tables:

```
CREATE TABLE `BalanceAccount` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `balance` int(11) NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=6 DEFAULT CHARSET=latin1
```

```
CREATE TABLE `BalanceTransaction` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `date` int(11) NOT NULL,
  `accountId` int(11) NOT NULL,
  `extraAccountId` int(11) NOT NULL DEFAULT '0',
  `amount` int(11) NOT NULL DEFAULT '0',
  `data` text,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=266 DEFAULT CHARSET=latin1

```

The component config:

```
        'balanceManager' => [
            'class' => 'yii2tech\balance\ManagerDb',
            'accountTable' => '{{%BalanceAccount}}',
            'transactionTable' => '{{%BalanceTransaction}}',
            'accountLinkAttribute' => 'accountId',
            'accountBalanceAttribute' => 'balance',
            'extraAccountLinkAttribute' => 'extraAccountId',
            'amountAttribute' => 'amount',
            'dataAttribute' => 'data',
            'autoCreateAccount' => true,
        ],
```

The transaction:

```
        // transfer :
        Yii::$app->balanceManager->transfer(
            [
                'id' => '5',
            ],
            [
                'id' => '4',
            ],
            100,
            // extra data associated with transaction :
            [
                'paymentGateway' => 'PayPal',
                'paymentId' => 'abcxyzerft',
            ]
        );
```

![query_01](https://cloud.githubusercontent.com/assets/10280718/16919118/b77fb6f2-4d08-11e6-854e-87b59476256e.png)
![query_02](https://cloud.githubusercontent.com/assets/10280718/16919126/c074c41e-4d08-11e6-8147-58f3f6b62d33.png)

**Comment by klimov-paul** [comment]
Resolved by commit f1db4719da77e9d915ebc7945da0cb54419e7cdb
